src/ensembl/src/shared/eye-toggle-icon/EyeToggleIcon.tsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import classNames from 'classnames';
+
+import eyeOnIcon from './eye-on.svg';
+import eyeOffIcon from './eye-off.svg';
+
+import styles from './EyeToggleIcon.scss';
+
+type Props = {
+  iconStatus: string;
+  description: string;
+  callbackFunction: () => void;
+};
+
+const EyeToggleIcon = (props: Props) => {
```

Comment:
I also have a naming concern. Is this component an "icon", or is it a "button" (since it has an `onClick` behavior)? @imran-ebi @ens-ap5 what would you say?

We are talking about this thingy, right?

![image](https://user-images.githubusercontent.com/6834224/54217033-c3377500-44e2-11e9-807a-1220edf39d2a.png)

Reply:
I would say that if it has behaviour, it's a button. An icon is an image used to represent a concept (rather than for its decorative effect).

Reply:
Since the component replaces the whole button instead of only the icon, it looks more like a "button" component to me.

Reply:
Again continuing on the subject of naming, wouldn't you agree that the name `VisibilityButton` better describes the purpose of this button than `EyeToggleButton`?

Reply:
I am not sure what the proper name should be. At the time of naming it, the purpose of it was not very clear to me as it did keep changing slightly now and then. But now it is surely more refined and we can confirm with Andrea what it is exactly for.

Reply:
I also noticed that we have the following buttons with similar functionality:

![Screenshot 2019-03-13 at 09 25 54](https://user-images.githubusercontent.com/3085815/54267987-30deb200-4572-11e9-8674-c29244e791fc.png)
![Screenshot 2019-03-13 at 09 25 34](https://user-images.githubusercontent.com/3085815/54267989-31774880-4572-11e9-8998-442d4ec7c676.png)

I guess it could all be implemented with just one component.

I feel `VisibilityButton` describes the purpose of the button but not the button itself.

What if we call it something like `ImageToggleButton` which would mean that it is an image that's used like a button and it can be toggled between on, off and disabled states?

Reply:
The track panel bar (vertical bar) buttons have a similar functionality. However, the launchbar icons belong to links so I am not sure if we can put them in the same category.

Reply:
The header buttons though can go into the same category.

![image](https://user-images.githubusercontent.com/32512909/54269186-cbd88b80-4574-11e9-9a29-186e7c48bef7.png)

Reply:
It might be worth having a single component if it consistently behaves the same (will it? I miss Andrea!). Only what will its interface (props) be?

Reply:
Won't we need to rely on a function for the link to do the redirection then? The link should be separate from the button for semantic reasons too in my opinion.

Reply:
If there is going to be a single button, I think we should start by prototyping its interface: what it requires from the outside world to function.

Something like:

```
OnClickProps = {
  onClick: () => void
} & CommonProps

LinkProps = {
  path: string
} & CommonProps

CommonProps = {
  ??? (is it going to be one `icon` or the `children` component)?
}
```

with

```
const VisibilityButton = () => {

  return (
    <ToggleImageButton onClick={blah} status={blah}>
      <EyeIcon/>
    </ToggleImageButton>
  );

}
```

or something?

I can understand why the wrapper component (like ToggleImageButton) may be useful, but on the other hand am not sure it will make our lives simpler.

Reply:
This looks interesting. Also, this made me wonder what if we have a generic icon component that has states. And we pass the icon information to it to render the eye icon rather than having a specific icon for it? A very abstract example would be:

`
const GenericStateIcon = ({ iconDetails }) => {
  return <img src={iconDetails.icon} />
}

const ToggleImageButton = () => {
  return <button><GenericStateIcon iconDetails={...} /></button>
}
`

Or the eye icon component can use this generic component instead?

`
const EyeIcon = () => {
  const iconDetails = {...}

  return <GenericStateIcon {...props} />
}

const ToggleImageButton = () => {
  return <button><EyeIcon /></button>
}
`

Reply:
> `return <img src={iconDetails.icon} />`

`svgr` though!

Also, now that you told me that a button can have more than 2 states, and that these 2 states can be visually represented by different (as opposed to just differently styled) svg's, I am a bit uncertain as to the practicality of this approach in general.

Of course, different svg's can still be put into 1 svg (as an svg sprite), and shown/hidden/styled using CSS.

Something we need to brainstorm...

Reply:
I get what you mean. This approach won't be suitable for SVG components, however can't we make these components configurable the same way?

Reply:
In the comment I actually meant to ask if we can have a generic wrapper component that will take in props on behalf of the SVG component, in this case the `EyeIcon`.